File: mobile-engage/src/main/java/com/emarsys/mobileengage/LoggingMobileEngageInternal.java
package com.emarsys.mobileengage;

import com.emarsys.core.api.result.CompletionListener;
import com.emarsys.core.util.SystemUtils;
import com.emarsys.core.util.log.Logger;
import com.emarsys.core.util.log.entry.MethodNotAllowed;

import java.util.HashMap;
import java.util.Map;

public class LoggingMobileEngageInternal implements MobileEngageInternal {

    private final Class klass;

    public LoggingMobileEngageInternal(Class klass) {
        this.klass = klass;
    }

    @Override
    public void setContact(String contactFieldValue, CompletionListener completionListener) {
        Map<String, Object> parameters = new HashMap<>();
        parameters.put("contact_field_value", contactFieldValue);
        parameters.put("completion_listener", completionListener != null);

        String callerMethodName = SystemUtils.getCallerMethodName();

        Logger.debug(new MethodNotAllowed(klass, callerMethodName, parameters));
    }

    @Override
    public void clearContact(CompletionListener completionListener) {
        Map<String, Object> parameters = new HashMap<>();
        parameters.put("completion_listener", completionListener);

        String callerMethodName = SystemUtils.getCallerMethodName();

        Logger.debug(new MethodNotAllowed(klass, callerMethodName, parameters));
    }
}

File: mobile-engage/src/main/java/com/emarsys/mobileengage/LoggingMobileEngageInternal.kt
package com.emarsys.mobileengage

import com.emarsys.core.api.result.CompletionListener
import com.emarsys.core.util.SystemUtils
import com.emarsys.core.util.log.Logger.Companion.debug
import com.emarsys.core.util.log.entry.MethodNotAllowed

class LoggingMobileEngageInternal(private val klass: Class<*>) : MobileEngageInternal {
    override fun setContact(
        contactFieldId: Int?,
        contactFieldValue: String?,
        completionListener: CompletionListener?
    ) {
        val parameters = mapOf(
            "contact_field_id" to contactFieldId,
            "contact_field_value" to contactFieldValue,
            "completion_listener" to (completionListener != null)
        )
        val callerMethodName = SystemUtils.getCallerMethodName()
        debug(MethodNotAllowed(klass, callerMethodName, parameters))
    }

    override fun setAuthenticatedContact(
        contactFieldId: Int,
        openIdToken: String,
        completionListener: CompletionListener?
    ) {
        val parameters = mapOf(
            "contact_field_id" to contactFieldId,
            "id_token" to openIdToken,
            "completion_listener" to (completionListener != null)
        )
        val callerMethodName = SystemUtils.getCallerMethodName()
        debug(MethodNotAllowed(klass, callerMethodName, parameters))
    }

    override fun clearContact(completionListener: CompletionListener?) {
        val parameters = mapOf(
            "completion_listener" to (completionListener != null)
        )
        val callerMethodName = SystemUtils.getCallerMethodName()
        debug(MethodNotAllowed(klass, callerMethodName, parameters))
    }
}